File: internal/content/testdata/bench/java/nc-kth-largest-stream.java
import java.util.*;

class Solution {
    public long[] kth_largest_stream_ops(long k, long[] nums, Object[][] operations) {
        PriorityQueue<Long> heap = new PriorityQueue<>(); // min-heap
        for (long n : nums) heap.add(n);
        while (heap.size() > k) heap.poll();
        long[] out = new long[operations.length];
        for (int i = 0; i < operations.length; i++) {
            Object[] op = operations[i];
            long val = ((Number) op[1]).longValue();
            heap.add(val);
            while (heap.size() > k) heap.poll();
            out[i] = heap.peek();
        }
        return out;
    }
}

File: internal/content/testdata/bench/csharp/nc-kth-largest-stream.cs
using System.Collections.Generic;

public class Solution {
    public long[] kth_largest_stream_ops(long k, long[] nums, object[][] operations) {
        var heap = new SortedList<long, int>(); // min-heap via SortedList with count
        // Use priority queue instead
        var pq = new PriorityQueue<long, long>(); // min-heap
        foreach (long n in nums) pq.Enqueue(n, n);
        while (pq.Count > k) pq.Dequeue();
        var out_ = new long[operations.Length];
        for (int i = 0; i < operations.Length; i++) {
            long val = (long)operations[i][1];
            pq.Enqueue(val, val);
            while (pq.Count > k) pq.Dequeue();
            out_[i] = pq.Peek();
        }
        return out_;
    }
}
